Fix missing DIMM temperature sensors

MoshiMoshi0 opened the pull request:
Fix dimm temperature sensors missing since merging #1797

The issue was mainly this check:
https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/06b714ae1b1a09fdbe866a5bd718bbdbb8fa5f55/LibreHardwareMonitorLib/Hardware/Memory/MemoryGroup.cs#L106-L109

Since `_opened` is false by default this will return true from `TryAddDimms` and not even attempt to add them. 
And also if `TryAddDimms` returns true the retry task will not be started.

Fixed by changing `_opened` to `_disposed` flag, also simplified `AddDimms` and `HardwareAdded` notification.

---

Might be out of scope of this PR but 
https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/06b714ae1b1a09fdbe866a5bd718bbdbb8fa5f55/LibreHardwareMonitorLib/Hardware/Memory/MemoryGroup.cs#L60
and
https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/blob/06b714ae1b1a09fdbe866a5bd718bbdbb8fa5f55/LibreHardwareMonitorLib/Hardware/Memory/MemoryGroup.cs#L71-L79
can cause "Collection was modified" exception since you can iterate on them outside of the lock.
I can fix this in this PR but `IReadOnlyList<IHardware> Hardware` would probably have to return a copy, or use some concurrent collection.

yw4z commented:
works as expected thanks

<img width="476" height="233" alt="Screenshot-20250815012022" src="https://github.com/user-attachments/assets/eb7a560d-aad2-4bc2-a39f-d09ec38ad096" />

Rem0o commented:
Thanks for the fix.

MoshiMoshi0 commented:
> Not really since a new list is assigned when new RAM is added.

Yea I removed the new list creation so that HardwareAdded handlers can access the hardware list with the new hardware included.
Reverted to creating a new list again in #1836 and invoking HardwareAdded after adding.
